### PR TITLE
feat: Support published status for landing pages and preview of draft pages

### PR DESCRIPTION
**Pull request by gidjin**
# SC-2577

## Proposed changes

Support showing a draft landing page banner on landing pages that are not published. Reword article draft banner to use consistent language. Update the landing page index to hide unpublished landing pages from users. Update landing page index to show draft and archived landing pages to cms users, and flagging them as `Draft` or `Archived`.

---

## Reviewer notes

As I noted in sync it was easier to support `Archived` landing pages than not. So that isn't part of the original AC but we did talk about it in the future so I included it. 

Design, please make sure I picked the right colors for the tag on the landing page index.

You will want https://github.com/USSF-ORBIT/ussf-portal-cms/pull/411 locally for this to fully work. By default all existing landing pages will be draft. You will need to publish one to fully test everything.

### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-cms
yarn dev
```

Login to the portal http://localhost:3000

### Start storybook

```sh
yarn storybook
```

Login to storybook http://localhost:6006, though the command above should open it for you

---

### As the original developer, I have

- [X] Met the acceptance criteria
- [X] Created new stories in Storybook if applicable
- [X] Created/modified automated unit tests in Jest
- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [X] Followed guidelines for zero-downtime deploys, if applicable
- [X] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

### View of draft landing page as CMS user
SCR-20231206-glx

### View of landing page index as CMS User
SCR-20231206-gmb

### View of landing page index as non CMS User
SCR-20231206-gnl

### View of storybook of landing page table component
SCR-20231206-mzl

#### Images showing really long title
SCR-20231207-h6s
SCR-20231207-h7c
SCR-20231207-h7l
SCR-20231207-h7r

### CODE REVIEW

File: src/pages/landing/[landingPage]/index.tsx [hunk 1]
```diff
@@ -131,6 +133,9 @@ const LandingPage = ({
 
   return (
     <Grid className={styles.inPageNav}>
+      {!isPublished(landingPage) && (
+        <h2 className={styles.previewBanner}>Draft Landing Page Preview</h2>
+      )}
```
[the remark this answers is not in the document]
Comment (gidjin) [reply src/pages/landing/[landingPage]/index.tsx:138]: 🤔 I could go either way. I'm not against adding it, and seems like a short lift, though that's not how articles work now and maybe we should do the same there too. Maybe that makes a good case to have it be a follow up story. I'll create one.

**Comment by AnnaGingle** [comment]
Hi there! The responsiveness is different than the [designs](https://www.figma.com/file/iPOdDkRxCaeoDoQ4JXjxlk/Space-Force-Service-Portal?type=design&node-id=18468-13154&mode=design&t=YXDNRryqfKEzwKli-4). 
Light mode

File: src/pages/landing/[landingPage]/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gidjin) [reply src/pages/landing/[landingPage]/index.tsx:138]: Story created https://app.shortcut.com/orbit-truss/story/2856

**Review by shkeating — COMMENTED** [review]
I think I might check out this branch and see if there is a way to make the different status tags be different colors to make it more obvious the pages are in different states

File: src/components/LandingPageIndexTable/LandingPageIndexTable.tsx [hunk 2]
```diff
@@ -19,12 +22,25 @@ const LandingPageIndexTable = ({
     <Table bordered striped className={`usa-table--borderless ${styles.table}`}>
       <tbody>
         {landingPages.map((landingPage: LandingPage) => {
+          const rowId = `landing_page_${landingPage.slug}`
+          // if a future publish date is set format the date for display
+          // we don't need to check if the date is in the future as the
+          // isPublished helper function will return false if the date is in the future
```
Comment (jcbcapps) [inline src/components/LandingPageIndexTable/LandingPageIndexTable.tsx:28]: Thanks for the explanation here!

File: src/pages/landing/index.tsx [hunk 3]
```diff
@@ -28,16 +27,38 @@ export default Landing
...
+  const sortableCopy = [...landingPages]
+  if (isCmsUser(user)) {
+    const sortedLandingPages = sortableCopy.sort(
+      (a: LandingPage, b: LandingPage) => a.pageTitle.localeCompare(b.pageTitle)
+    )
+    return {
+      props: {
+        landingPages: sortedLandingPages,
+      },
+    }
```
Comment (jcbcapps) [inline src/pages/landing/index.tsx:50]: Good call moving the sorting here. I think this is a better place for it.

**Review by jcbcapps — APPROVED** [review]
LGTM

File: src/pages/landing/[landingPage]/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (abbyoung) [reply src/pages/landing/[landingPage]/index.tsx:138]: Agree that an 'Archived' banner will make more sense. Thanks for creating that story!

**Comment by abbyoung** [comment]
This is great! I had a couple takeaways/questions but I do not think any of them are blockers.

* If there are existing landing pages, when the migration is applied, they will be given the 'Draft' status and be unpublished. (This is what happened to me locally, at least.) We don't have any published on prod so this should be fine, but we'll want to make sure and update the pages on staging.
* If an article is published, and then put back to draft, the published date does not change. Does that matter at all? Maybe a UX question for future consideration.
* Archived pages show the 'Draft' banner, which you already created a story for future work.

**Comment by shkeating** [comment]
added colors to the tags per design feedback @AnnaGingle :)

**Comment by shkeating** [comment]
@gidjin is there a reason we want to hide published tags? I see there is tests for it that are now failing because I removed that logic that would hide them

**Comment by gidjin** [comment]
> @gidjin is there a reason we want to hide published tags? I see there is tests for it that are now failing because I removed that logic that would hide them

No. Labels were added for draft so I didn't think to show them for published also. Would they only show for CMS users?

**Comment by gidjin** [comment]
Here is a screenshot showing it on all of them. I also made it use the correct tense if publishing a future date or past date. Thoughts on which we should go with? Maybe tomorrow at sync we can discuss. SCR-20231213-l13

**Comment by gidjin** [comment]
as discussed I hide the labels from portal users and only display them for cms users. 

SCR-20231214-fzd

**Review by abbyoung — APPROVED** [review]

**Comment by AnnaGingle** [comment]
Looks good to me!

**Review by AnnaGingle — APPROVED** [review]

**Pull request merged**